core/unit_test/TestGraph.hpp

```diff
@@ -398,6 +398,16 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), empty_graph) {
   ex.fence();
 }
 
+// Ensure that an empty graph on the default host execution space
+// can be submitted.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), empty_graph_default_host_exec) {
```

Comment:
Skip if default host exec is the same as default exec?

Also why is this part of this PR?

Reply:
I wrote this test when I realized that for backends such as `HIP` or `SYCL`, `Kokkos::Graph<backend>` could potentially go on the defaulted implementation. My main concern was that at some point we could potentially make things wrong enough such that it would not be possible anymore to instantiate the graph for the default host execution space type.